design/pv_backup_info.md

```diff
@@ -0,0 +1,155 @@
+
+The `restoreItem` function can decode the _backup-name_-volumes-info.json file to determine how to handle the PV resource. 
+
+## Detailed Design
+
+### The VolumeInfo structure
+_backup-name_-volumes-info.json file is an array of structure `VolumeInfo`.
+
+The `VolumeInfo` definition is:
+``` golang
+type VolumeInfo struct {
+    PVCName        string    // The PVC's name. The format should be <namespace-name>/<PVC-name>
+    PVName         string    // The PV name.
+    BackupMethod   string    // The way the volume data is backed up. The valid value includes `VeleroNativeSnapshot`, `PodVolumeBackup`, `CSISnapshot`, `SnapshotDataMover`, and `Skipped`.
+
+    SkippedReason  string    // The reason for the volume is skipped in the backup.
+
+    SnapshotHandle      string       // The actual snapshot ID. It can be the cloud provider's snapshot ID or the file-system uploader's snapshot.
+    Status              string       // The snapshot's final status.
+    Size                int64        // The snapshot corresponding volume size. Some of the volume backup methods cannot retrieve the data by current design, for example, the Velero native snapshot.
+    StartTimestamp      *metav1.Time // Snapshot starts timestamp.
+    CompletionTimestamp *metav1.Time // Snapshot completes timestamp.
+
+    DataMoverInfo       DataMoverInfo
+    NativeSnapshotInfo  VeleroNativeSnapshotInfo
+    PVBInfo             PodVolumeBackupInfo
+}
+
+// DataMoverInfo is used for displaying the snapshot data mover and PodVolumeBackup snapshot status.
+type DataMoverInfo struct {
+    DataMover        string    // The name of the data mover that uploads the snapshot data. The valid values are `kopia` and `restic`. It's useful for file-system backup and snapshot data mover.
+    RetainedSnapshot string    // The ID of the local snapshot.
+}
+
+// VeleroNativeSnapshotInfo is used for displaying the Velero native snapshot status.
+type VeleroNativeSnapshotInfo struct {
+    VolumeType string    // The cloud provider snapshot volume type.
+    VolumeAZ   string    // The cloud provider snapshot volume's availability zones.
+    IOPS       string    // The cloud provider snapshot volume's IOPS.
+}
+
+// PodVolumeBackupInfo is used for displaying the PodVolumeBackup snapshot status.
+type PodVolumeBackupInfo struct {
+    VolumeName string // The PVC's corresponding volume name used by Pod
+    PodName    string // The Pod name mounting this PVC. The format should be <namespace-name>/<pod-name>.
+}
+```
+
+### How the VolumeInfo array is generated.
+
+Two new methods are added to BackupStore for uploading and downloading the VolumeInfo metadata file.
+
+``` golang
+type BackupStore interface {
+    ...
+    PutVolumeInfos(backup string, volumeInfos io.Reader) error
+    GetVolumeInfos(name string) ([]*VolumeInfo, error)
+    ...
+}
+```
+
+There are two phases to handle the VolumeInfo array.
```

Comment:
Let's also describe when and how this data is deleted from the backup repo.

Reply:
The metadata file is deleted when the backup is deleted.
I will document it in the design.